File: academic-system/src/test/java/org/example/model/TurmaIgualdadeTeste.java
package org.example.model;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class TurmaIgualdadeTeste {

    @Test
    public void turmasComMesmoCodigoDevemSerIguais() {
        Turma t1 = new Turma("POO", "Programação Orientada a Objetos");
        Turma t2 = new Turma("POO", "Outro Título");
        
        assertEquals(t1, t2);
        assertEquals(t1.hashCode(), t2.hashCode());
    }

    @Test
    public void turmasComCodigosDiferentesDevemSerDiferentes() {
        Turma t1 = new Turma("POO1", "Programação Orientada a Objetos");
        Turma t2 = new Turma("POO2", "Programação Orientada a Objetos");
        
        assertNotEquals(t1, t2);
    }
}

File: academic-system-csharp/tests/SistemaAcademico.Tests/Models/TurmaIgualdadeTeste.cs
using Xunit;
using SistemaAcademico.Models;

namespace SistemaAcademico.Tests.Models;

// Equivalente à classe Java: TurmaIgualdadeTeste
public class TurmaIgualdadeTeste
{
    [Fact]
    public void TurmasComMesmoCodigoDevemSerIguais()
    {
        Turma t1 = new Turma("POO", "Programação Orientada a Objetos");
        Turma t2 = new Turma("POO", "Outro Título");

        Assert.Equal(t1, t2);
        Assert.Equal(t1.GetHashCode(), t2.GetHashCode());
    }

    [Fact]
    public void TurmasComCodigosDiferentesDevemSerDiferentes()
    {
        Turma t1 = new Turma("POO1", "Programação Orientada a Objetos");
        Turma t2 = new Turma("POO2", "Programação Orientada a Objetos");

        Assert.NotEqual(t1, t2);
    }
}
